src/app/ApplicationTemplate.Shared/Presentation/Authentication/CreateAccountFormViewModel.cs

```diff
@@ -20,19 +43,111 @@ public string Email
 			set => this.Set(value);
 		}
 
+		public string PhoneNumber
+		{
+			get => this.Get<string>();
+			set => this.Set(value);
+		}
+
+		public string SecondaryPhoneNumber
+		{
+			get => this.Get<string>();
+			set => this.Set(value);
+		}
+
+		public string PostalCode
+		{
+			get => this.Get<string>();
+			set => this.Set(value);
+		}
+
+		public DateTimeOffset DateOfBirth
+		{
+			get => this.Get<DateTimeOffset>(initialValue: DateTimeOffset.Now);
+			set => this.Set(value);
+		}
+
 		public string Password
 		{
 			get => this.Get<string>();
 			set => this.Set(value);
 		}
+
+		public string ConfirmPassword
+		{
+			get => this.Get<string>();
+			set => this.Set(value);
+		}
+
+		public string[] FavoriteDonuts
+		{
+			get => this.Get<string[]>();
+			set => this.Set(value);
+		}
+
+		public bool AgreeToTermsOfServices
+		{
+			get => this.Get<bool>();
+			set => this.Set(value);
+		}
 	}
 
 	public class CreateAccountFormValidator : AbstractValidator<CreateAccountFormViewModel>
 	{
-		public CreateAccountFormValidator()
+		public CreateAccountFormValidator(IStringLocalizer localizer)
 		{
+			var myPassword = string.Empty;
+
+			RuleFor(x => x.FirstName).NotEmpty();
+			RuleFor(x => x.LastName).NotEmpty();
 			RuleFor(x => x.Email).NotEmpty().EmailAddress();
-			RuleFor(x => x.Password).NotEmpty();
+			RuleFor(x => x.PhoneNumber).NotEmpty().Length(14).WithMessage(localizer["CreateAccount_PhoneNumberValidation"]); // "(000) 000-0000".Length = 14
+			RuleFor(x => x.SecondaryPhoneNumber).Must(phonenumber =>
+			{
+				if (phonenumber.IsNullOrEmpty())
+				{
+					return true;
+				}
+				else if (phonenumber.Length == 14)
```

Comment:
there are multiple ways of writing a phone number. You shouldn't check the string length, but rather its content.

Reply:
but in our case the the phone number is strongly formatted, only numbers and other characters are added automatically.

Reply:
specified expected formatting in the titles of phone numbers